magento2-nginx/fpm-7.0/usr/local/share/magento2/magento_functions.sh

```diff
@@ -0,0 +1,245 @@
+      chown -R "${CODE_OWNER}:${CODE_GROUP}" pub/static/frontend/
+    fi
+
+    if [ ! -d "$FRONTEND_INSTALL_DIRECTORY/node_modules" ]; then
+      as_code_owner "npm install" "$FRONTEND_INSTALL_DIRECTORY"
+    fi
+    if [ -z "$GULP_BUILD_THEME_NAME" ]; then
+      as_code_owner "gulp $FRONTEND_BUILD_ACTION" "$FRONTEND_BUILD_DIRECTORY"
+    else
+      as_code_owner "gulp $FRONTEND_BUILD_ACTION --theme='$GULP_BUILD_THEME_NAME'" "$FRONTEND_BUILD_DIRECTORY"
+    fi
+
+  if [ -d "pub/static/frontend/" ] && [ "$IS_CHOWN_FORBIDDEN" != 'true' ]; then
+      chown -R "${APP_USER}:${APP_GROUP}" pub/static/frontend/
+    fi
+  fi
+}
+
+function do_magento_install_custom() {
+  if [ -f "/usr/local/share/magento2/install_magento_custom.sh" ]; then
+    # shellcheck source=./install_magento_custom.sh
+    source "/usr/local/share/magento2/install_magento_custom.sh"
+  fi
+}
+
+function do_magento_switch_web_writable_directories_to_code_owner() {
+  if [ "$IS_CHOWN_FORBIDDEN" != 'true' ]; then
+    chown -R "${CODE_OWNER}":"${CODE_GROUP}" pub/media pub/static var
+  else
+    chmod a+rw pub/media pub/static var
+  fi
+}
+
+function do_magento_move_compiled_assets_away_from_codebase() {
+  # Preserve compiled theme files across setup:upgrade calls.
+  if [ -d pub/static/frontend/ ]; then
+    mkdir /tmp/assets
+    mv pub/static/frontend/ /tmp/assets/
+  fi
+}
+
+function do_magento_setup_upgrade() {
+  rm -rf var/generation/*
+  redis-cli -h "$REDIS_HOST" -p "$REDIS_HOST_PORT" -n "$MAGENTO_REDIS_CACHE_DATABASE" "FLUSHDB"
+  redis-cli -h "$REDIS_HOST" -p "$REDIS_HOST_PORT" -n "$MAGENTO_REDIS_FULL_PAGE_CACHE_DATABASE" "FLUSHDB"
+  as_code_owner "bin/magento setup:upgrade"
+}
+
+function do_magento_move_compiled_assets_back_to_codebase() {
+  if [ -d /tmp/assets/ ]; then
+    mkdir -p pub/static/
+    mv /tmp/assets/frontend/ pub/static/
+    rm -rf /tmp/assets
+  fi
+}
+
+function do_magento_dependency_injection_compilation() {
+  # Compile the DIC if to be productionized
+  if [ "$PRODUCTION_ENVIRONMENT" = "1" ]; then
+    as_code_owner "$MAGENTO_DEPENDENCY_INJECTION_COMPILE_COMMAND"
+  fi
+}
+
+function do_magento_deploy_static_content() {
+  # Compile static content if it's a production container.
+  if [ "$MAGENTO_MODE" = "production" ]; then
+    as_code_owner "bin/magento setup:static-content:deploy $FRONTEND_COMPILE_LANGUAGES"
+  fi
+}
+
+function do_magento_reindex() {
+  (as_code_owner "bin/magento indexer:reindex" || echo "Failing indexing to the end, ignoring.") && echo "Indexing successful"
+}
+
+function do_magento_assets_download() {
+  if [ "$IS_HEM" == 'true' ]; then
+    export HEM_RUN_ENV="${HEM_RUN_ENV:-local}"
+    for asset_env in $ASSET_DOWNLOAD_ENVIRONMENTS; do
+      as_build "hem --non-interactive --skip-host-checks assets download -e $asset_env"
+    done
+  fi
+}
+
+function do_magento_cache_flush() {
+  # Flush magento cache
+  as_code_owner "bin/magento cache:flush"
+}
+
+function do_magento_create_web_writable_directories() {
```

Comment:
This double version of do_magento_create_web_writable_directories (see line 18) stops bin/magento from working.